internal/services/kusto/kusto_iothub_data_connection_resource.go

```diff
@@ -217,13 +226,44 @@ func resourceKustoIotHubDataConnectionRead(d *pluginsdk.ResourceData, meta inter
 				d.Set("database_routing_type", string(pointer.From(props.DatabaseRouting)))
 				d.Set("shared_access_policy_name", props.SharedAccessPolicyName)
 				d.Set("event_system_properties", utils.FlattenStringSlice(props.EventSystemProperties))
+				d.Set("retrieval_start_date", pointer.From(props.RetrievalStartDate))
 			}
 		}
 	}
 
 	return nil
 }
 
+func resourceKustoIotHubDataConnectionUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Kusto.DataConnectionsClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := dataconnections.ParseDataConnectionID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	iotHubDataConnectionProperties := expandKustoIotHubDataConnectionProperties(d)
+
+	dataConnection := dataconnections.IotHubDataConnection{
+		Location:   pointer.To(location.Normalize(d.Get("location").(string))),
+		Properties: iotHubDataConnectionProperties,
+	}
+
+	if databaseRouting, ok := d.GetOk("database_routing_type"); ok {
```

Comment:
`database_routing_type` isn't updatable since it's ForceNew in the schema but if it is updatable, let's get that fixed in the schema.

Reply:
Will fix, thx.